mlflow/store/artifact/http_artifact_repo.py

```diff
@@ -1,32 +1,30 @@
 import os
-import requests
 import posixpath
 
 from mlflow.entities import FileInfo
 from mlflow.store.artifact.artifact_repo import ArtifactRepository, verify_artifact_path
 from mlflow.utils.file_utils import relative_path_to_artifact_path
-from mlflow.utils.rest_utils import augmented_raise_for_status
+from mlflow.utils.rest_utils import MlflowHostCreds, augmented_raise_for_status, http_request
 
 
 class HttpArtifactRepository(ArtifactRepository):
     """Stores artifacts in a remote artifact storage using HTTP requests"""
 
     def __init__(self, artifact_uri):
         super().__init__(artifact_uri)
-        self._session = requests.Session()
+        self.host_creds = self._get_host_creds(artifact_uri)
 
-    def __del__(self):
-        if hasattr(self, "_session"):
-            self._session.close()
+    def _get_host_creds(self, uri):
+        return MlflowHostCreds(uri)
```

Comment:
@TimNooren Can we use `_get_default_host_creds()` here to enable users to easily configure auth for artifact operations initiated through `HttpArtifactRepository`? This is the current behavior when constructing `RestStore` via https://github.com/mlflow/mlflow/blob/70fc2dd41162156d11f6070b2e75a478db7ad627/mlflow/tracking/_tracking_service/utils.py#L133, so I think it makes sense to be consistent here.

Reply:
Ah, I see. My understanding was that the `HttpArtifactRepository` is intended to be agnostic to actual service being used to handle artifacts, so we should not assume anything about where it is running or how it is secured. But I guess we can assume it's at least behind the same ingress as the tracking server and uses the same credentials. Or maybe I just misinterpreted the purpose altogether:) Anyway I will update this👍

Reply:
I got rid of the indirection through `_get_host_creds`, since both `HttpArtifactRepository` and `MlflowArtifactsRepository` will be getting credentials through `_get_default_host_creds`. There is no need for overriding the behaviour in `MlflowArtifactsRepository`:)